osu.Game/Screens/Select/FilterControl.cs

```diff
@@ -214,8 +217,14 @@ private void load(OsuColour colours, IBindable<RulesetInfo> parentRuleset, OsuCo
             config.BindWith(OsuSetting.DisplayStarsMaximum, maximumStars);
             maximumStars.ValueChanged += _ => updateCriteria();
 
-            ruleset.BindTo(parentRuleset);
             ruleset.BindValueChanged(_ => updateCriteria());
+            mods.BindValueChanged(_ =>
+            {
+                // Mods are updated once by the mod select overlay when song select is entered, regardless of if there are any mods.
+                // Updating the criteria here so early triggers a re-ordering of panels on song select, via... some mechanism.
+                // Todo: Investigate/fix the above and remove this schedule.
```

Comment:
> I guess this looks mostly tolerable, but tests are failing, and you [seem to know the reason why](https://discord.com/channels/188630481301012481/188630652340404224/1222949671635521607)?

This is the reason for the tests breaking, because they assert the number of filter operations that have taken place. Turns out that a local `SequenceEqual()` works better than the schedule.

Reply:
If it's known why this entire thing is required and the `SequenceEqual()` is final, then the todo comment probably needs to be revised / removed?

Reply:
That depends on whether the `SequenceEqual()` should be moved elsewhere. I wouldn't expect this to be called once with `OldValue: { NoFail }, NewValue: { NoFail }` once when song select is entered (or empty collections, or any other _but matching_ values).  
On the other hand, I also wouldn't necessarily expect `Bindable<T>` to automatically do a `SequenceEqual` if T is a collection type. Perhaps an equality comparer could be provided to the Bindable ctor for this particular case (similar to how you can do the same for `Dictionary<>` etc).

But yeah, I still consider this a bit of a hack for now, which is why I've left the todo in.

Reply:
> That depends on whether the `SequenceEqual()` should be moved elsewhere.

I don't think it should be and it's unlikely to ever happen unless we basically rewrite global mod handling to not require it (and probably half the game alongside). It's basically a weird situation spurred by the mod select overlay wanting to be in control of every mod _instance_ in the global bindable so that its changes propagate everywhere else correctly.

But I dunno. I'm already frustrated from feeling like I can't get anything merged due to stupid details so I might just look away.